include/exiv2/jpgimage.hpp

```diff
@@ -49,11 +49,11 @@ namespace Exiv2 {
      */
     struct EXIV2API Photoshop {
         // Todo: Public for now
-        static const char* const ps3Id_;                 //!< %Photoshop marker
-        static const std::array<const char*, 4> irbId_;  //!< %Photoshop IRB markers
-        static const char* const bimId_;                 //!< %Photoshop IRB marker (deprecated)
-        static const uint16_t iptc_;    //!< %Photoshop IPTC marker
-        static const uint16_t preview_; //!< %Photoshop preview marker
+        inline static const char* const ps3Id_ = "Photoshop 3.0\0"; //!< %Photoshop marker
```

Comment:
AFAIK inline here is redundant. static implies inline. This should also be constexpr.

Reply:
Maybe even std::string_view. Although I assume that would break API.

Reply:
I do not think `static` implies `inline`. The `inline` specified for static variable members is something new in C++17:
https://en.cppreference.com/w/cpp/language/inline

I just did this change because the previous code was failing to compile with C++17.

Reply:
it's constexpr that implies inline.

what compile error?

Reply:
I just checked it out. It was actually not a compiler error but linking one:

```
-- Build files have been written to: /media/linuxDev/programming/exiv2/buildRelease
[134/183] Linking CXX executable bin/key-test
FAILED: bin/key-test 
: && ccache /usr/bin/c++  -O3 -DNDEBUG  -fstack-protector-strong samples/CMakeFiles/key-test.dir/key-test.cpp.o  -o bin/key-test  -Wl,-rpath,/media/linuxDev/programming/exiv2/buildRelease/lib  lib/libexiv2.so.1.0.0.9  /home/luis/.conan/data/zlib/1.2.11/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/lib/libz.a && :
/usr/bin/ld: lib/libexiv2.so.1.0.0.9: undefined reference to `Exiv2::Photoshop::irbId_'
```
I think the problem is that in the header file it was declared as:
```
static const std::array<const char*, 4> irbId_;
```
While in the cpp file:
```
constexpr std::array<const char*, 4> Photoshop::irbId_
```

I could also declare it as `constexpr` in the header file as you suggested so that we do not need to add the `inline` specifier.

Reply:
No that error has to do with Windows and dllexport. defining them in the header is the correct solution.

Small nitpick: static constexpr vs. constexpr static.

Reply:
I actually got the error on Ubuntu 20.4 with Gcc-10 :sweat_smile: . I'll change the order to `static constexpr`

Reply:
I just realized. C++17 supports CTAD. You can safely ommit the template parameters.